hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestHoodieTableValuedFunction.scala

```diff
@@ -558,4 +558,50 @@ class TestHoodieTableValuedFunction extends HoodieSparkSqlTestBase {
       }
     }
   }
+
+  test(s"Test hudi_metadata Table-Valued Function") {
+    if (HoodieSparkUtils.gteqSpark3_2) {
+      withTempDir { tmp =>
+        Seq("cow").foreach { tableType =>
```

Comment:
let's write down a few tests with MOR table and compaction/clustering after 2 commits.